controllers/feeds/reactionList.js

```diff
@@ -20,9 +22,29 @@ module.exports = async (req, res) => {
       return {...commentRes, latest_children: {comment: sortComment}};
     });
 
+    // get comment users
+
+    let actors = [];
+    const collectUserIds = (comment) => {
+      if (comment?.user?.id && !actors.includes(comment.user.id)) {
+        actors.push(comment.user.id);
+      }
+      comment?.latest_children?.comment?.forEach(collectUserIds);
+    };
+    sortByDate.forEach(collectUserIds);
+
+    const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
+
     const removeSensitiveData = await Promise.all(
       sortByDate.map(async (data) => {
-        return handleAnonymousData(data, req, post.actor.id, myAnonymousId, data.user_id);
+        return handleAnonymousData(
+          data,
+          req,
+          post.actor.id,
+          myAnonymousId,
+          data.user_id,
+          karmaScores
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for collecting user IDs and getting their karma scores seems fine. However, I would suggest moving the logic of collecting user IDs into a separate function for better readability and maintainability. Also, it's important to handle potential errors that might occur when calling `UsersFunction.getUsersKarmaScore(User, actors)`. If this function fails for any reason, it could cause the entire endpoint to fail. Consider wrapping this call in a try-catch block.
```diff
     let actors = [];
     const collectUserIds = (comment) => {
       if (comment?.user?.id && !actors.includes(comment.user.id)) {
         actors.push(comment.user.id);
       }
       comment?.latest_children?.comment?.forEach(collectUserIds);
     };
     sortByDate.forEach(collectUserIds);

+    let karmaScores;
+    try {
+      karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
+    } catch (error) {
+      console.error('Failed to get users karma scores:', error);
+      // Handle or throw the error as per your application's error handling strategy
+    }

     const removeSensitiveData = await Promise.all(
       sortByDate.map(async (data) => {
        return handleAnonymousData(
          data,
          req,
          post.actor.id,
          myAnonymousId,
          data.user_id,
          karmaScores
        );
      })
    );
    let removeDeletedUser = removeSensitiveData.filter((data) => data.user);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->